cmd/vsphere-xcopy-volume-populator/e2e-tests/ansible/setup-vm.yml

```diff
@@ -0,0 +1,292 @@
+        VSPHERE_PASSWORD: "{{ vsphere_password }}"
+        VSPHERE_VERIFY_SSL: "{{ lookup('env', 'VSPHERE_VERIFY_SSL') | default('true') }}"
+      register: datastore_info_result
+      changed_when: false
+
+    - name: Set datastore facts
+      ansible.builtin.set_fact:
+        datastore_name: "{{ (datastore_info_result.stdout | from_json).name }}"
+        datastore_id: "{{ (datastore_info_result.stdout | from_json).id }}"
+      when: datastore_info_result.rc == 0
+
+    - name: Fail if datastore not found
+      ansible.builtin.fail:
+        msg: "Failed to resolve datastore: {{ datastore_info_result.stderr | from_json | default({'error': 'Unknown error'}) }}"
+      when: datastore_info_result.rc != 0
+
+
+    - name: Create VM
+      community.vmware.vmware_guest:
+        hostname: "{{ vsphere_host }}"
+        username: "{{ vsphere_username }}"
+        password: "{{ vsphere_password }}"
+        validate_certs: "{{ vsphere_validate_certs }}"
+        datacenter: "{{ datacenter }}"
+        cluster: "{{ cluster | default(omit) }}"
+        folder: "{{ vm_folder }}"
+        name: "{{ vm_name }}"
+        state: present
+        template: "{{ vm_template if vm_template else omit }}"
+        guest_id: "{{ vm_guest_id if not vm_template else omit }}"
+        convert: "{{ vm_disk_type if vm_template else omit }}"
+        datastore: "{{ datastore_name if vm_template else omit }}"
+        disk: "{{ disk_spec_no_template if not vm_template else omit }}"
+        hardware:
+          memory_mb: "{{ vm_memory }}"
+          num_cpus: "{{ vm_cpu }}"
+          boot_firmware: "{{ 'bios' if not vm_template else omit }}"
+        networks:
+          - name: "{{ network }}"
+            device_type: vmxnet3
+        cdrom: "{{ [{'controller_number': 0, 'unit_number': 0, 'type': 'iso', 'iso_path': vm_iso_path, 'state': 'present'}] if vm_iso_path else omit }}"
+      no_log: true
+      vars:
+        disk_spec_no_template:
+          - size_gb: "{{ vm_disk_size }}"
+            type: "{{ vm_disk_type }}"
+            datastore: "{{ datastore_name }}"
+
+    - name: Power on and get VM info
+      block:
+        - name: Power on VM
+          community.vmware.vmware_guest_powerstate:
+            hostname: "{{ vsphere_host }}"
+            username: "{{ vsphere_username }}"
+            password: "{{ vsphere_password }}"
+            validate_certs: "{{ vsphere_validate_certs }}"
+            name: "{{ vm_name }}"
+            state: powered-on
+          no_log: true
+
+        - name: Get VM info
+          community.vmware.vmware_guest_info:
+            hostname: "{{ vsphere_host }}"
+            username: "{{ vsphere_username }}"
+            password: "{{ vsphere_password }}"
+            validate_certs: "{{ vsphere_validate_certs }}"
+            name: "{{ vm_name }}"
+            datacenter: "{{ datacenter }}"
+          no_log: true
+          register: vm_info
+
+    - name: Wait for SSH to become available and add host key
```

Comment:
I think all of this SSH setup can be saved by using cloudinit,

Reply:
if we create a vm from template this would mean only doing this once

Reply:
actually, I'm not sure why we even need this ssh setup in the vm, I think I added it because of code rabbit suggestion, but booting a vm from template should just be enough, we don't need something special inside the vm for the migration, right?